typescript/agentkit/src/wallet-providers/svmKeypairWalletProvider.ts

```diff
@@ -0,0 +1,110 @@
+
+export class SvmKeypairWalletProvider extends SvmWalletProvider {
+    #keypair: Keypair;
+    #connection: Connection;
+    #genesisHash: string;
+
+    /**
+     * Creates a new SvmKeypairWalletProvider
+     * @param keypair - Either a Uint8Array or a base58 encoded string representing a 32-byte secret key
+     * @param rpcUrl - URL of the Solana RPC endpoint
+     */
+    constructor({
+        keypair,
+        rpcUrl,
+        genesisHash,
+    }: {
+        keypair: Uint8Array | string,
+        rpcUrl: string,
+        genesisHash: string,
+    }) {
+        super();
+
+        this.#keypair = typeof keypair === "string" ? Keypair.fromSecretKey(bs58.decode(keypair)) : Keypair.fromSecretKey(keypair);
+        this.#connection = new Connection(rpcUrl);
+        this.#genesisHash = genesisHash;
+    }
+
+    getAddress(): string {
+        return this.#keypair.publicKey.toBase58();
+    }
+
+    getNetwork(): Network {
+        if (this.#genesisHash === SOLANA_MAINNET_GENESIS_BLOCK_HASH) {
+            return {
+                protocolFamily: SOLANA_PROTOCOL_FAMILY,
+                chainId: String(SOLANA_MAINNET_CHAIN_ID),
+                networkId: SOLANA_MAINNET_NETWORK_ID,
+            };
+        } else if (this.#genesisHash === SOLANA_TESTNET_GENESIS_BLOCK_HASH) {
+            return {
+                protocolFamily: SOLANA_PROTOCOL_FAMILY,
+                chainId: String(SOLANA_TESTNET_CHAIN_ID),
+                networkId: SOLANA_TESTNET_NETWORK_ID,
+            };
+        } else if (this.#genesisHash === SOLANA_DEVNET_GENESIS_BLOCK_HASH) {
+            return {
+                protocolFamily: SOLANA_PROTOCOL_FAMILY,
+                chainId: String(SOLANA_DEVNET_CHAIN_ID),
+                networkId: SOLANA_DEVNET_NETWORK_ID,
+            };
+        } else {
+            throw new Error(`Unknown network with genesis hash: ${this.#genesisHash}`);
+        }
+    }
+
+    signTransaction(transaction: VersionedTransaction): VersionedTransaction {
+        transaction.sign([this.#keypair])
+        return transaction
+    }
+
+    sendTransaction(transaction: VersionedTransaction): Promise<string> {
+        return this.#connection.sendTransaction(transaction);
+    }
+
+    waitForTransactionReceipt(txHash: string): Promise<any> {
+        return this.#connection.confirmTransaction(txHash);
+    }
+
+    getName(): string {
+        return "svm_keypair_wallet_provider";
+    }
+
+    getBalance(): Promise<bigint> {
+        return this.#connection.getBalance(this.#keypair.publicKey).then(balance => BigInt(balance))
+    }
+
+    // Assumes `to` is a hex encoded address that we'll convert to a Solana PublicKey
+    async nativeTransfer(to: `0x${string}`, value: string): Promise<`0x${string}`> {
+        const toPubkey = new PublicKey(Buffer.from(to.slice(2), "hex"));
```

Comment:
We will weaken the `WalletProvider#nativeTransfer` abstract method signature to support `string` instead of `0x${string}`. It doesn't break anything explicitly yet to weaken the type and I would prefer to have solana feel native from day 0